feat: remove depdendecy on @types/node

wesleytodd opened the pull request:
I think this is a worthy change to reduce installing `@types/node` for the cases that dont need it. cc @boneskull since you wrote these, tell me if you think this is a bad idea.

wesleytodd committed:
feat: remove depdendecy on @types/node

boneskull reviewed (approved):
seems reasonable to me

remcohaszing committed:
Fix type issues

This change fixes various type issues on top of #24.

The `declare module` block is removed. It’s unnecessary and generally
discouraged. A good use case is to declare additional modules that don’t
match the package name, for example `@types/node` declaring `fs`.

This package exports a value by assigning `module.exports`. The correct
way to type this is using the `export =` syntax. Previously the types
had a default export. A default export in CJS types, represents the
value of `module.exports.default`. This property does not exist. In
order to export additional types, a namespace is used.

The type `LoggerrStatic` was only used to extend from by
`LoggerrConstructor`. It’s properties were merged into
`LoggerrContructor` and the interface is removed.

The `DefaultStreams` tuple had comments to label severities. This has
been replaced with real labels. This has no impact besides better editor
support.

The `Loggerr_` class was unused. It’s definition was merged into the
`Loggerr` type.

A proper `tsconfig.json` was added to test the types.

The types tests were modified to work with modern module resolution
options and `useVerbatimModuleSyntax`.

The type definitions file was renamed to match the main file. This was
editors provide some level of type support while working on the source
files locally. This also makes the `types` field in `package.json`
redundant.

wesleytodd committed:
fix(test): fixed import in ts test
